community/kernel/src/main/java/org/neo4j/kernel/impl/index/schema/NativeSchemaNumberIndexProvider.java

```diff
@@ -115,14 +121,35 @@ public String getPopulationFailure( long indexId ) throws IllegalStateException
     private String readPopulationFailure( long indexId ) throws IOException
     {
         NativeSchemaIndexHeaderReader headerReader = new NativeSchemaIndexHeaderReader();
-        GBPTree.readHeader( pageCache, nativeIndexFileFromIndexId( indexId ), new ReadOnlyMetaNumberLayout(), headerReader );
+        GBPTree.readHeader( pageCache, nativeIndexFileFromIndexId( indexId ), new ReadOnlyMetaNumberLayout(),
+                headerReader );
         return headerReader.failureMessage;
     }
 
     @Override
     public InternalIndexState getInitialState( long indexId, IndexDescriptor descriptor )
     {
-        return null;
+        try
+        {
+            NativeSchemaIndexHeaderReader headerReader = new NativeSchemaIndexHeaderReader();
+            GBPTree.readHeader( pageCache, nativeIndexFileFromIndexId( indexId ), new ReadOnlyMetaNumberLayout(),
+                    headerReader );
+            switch ( headerReader.state )
+            {
+            case BYTE_FAILED:
+                return InternalIndexState.FAILED;
+            case BYTE_ONLINE:
+                return InternalIndexState.ONLINE;
+            case BYTE_POPULATING:
+                return InternalIndexState.POPULATING;
+            default:
+                throw new IllegalStateException( "Unexpected initial state byte value " + headerReader.state );
+            }
+        }
+        catch ( IOException e )
+        {
+            return InternalIndexState.POPULATING;
```

Comment:
`LuceneSchemaIndexProvider` also do `log.error( "Failed to open index:" + indexId + ", requesting re-population.", e );`. Maybe native should as well?

Also, missing test for this behaviour.